### ISSUE TITLE
Move serialized data to Json format

### ISSUE THREAD

**Issue by romainruaud** [Smile-SA/elasticsuite#916]
According to https://devdocs.magento.com/guides/v2.2/release-notes/backward-incompatible-changes/

Already done on virtual categories rule storage, but should also be done on Optimizer config & rule.

Additional check has to be performed on the rest of the module to check serialize/unserialize potential usage.

**Comment by afoucret** [comment]
From the doc :

> Case 4:
> The serialize/unserialize logic belongs to your extension and stores data in a new table or field.
> Solution:
> Your extension will continue working normally, but we recommend switching to JSON for security reasons. You also need to write an upgrade script for the data in the database.

Not sure if we will be affected by this one. But in the long term, moving to JSON should be done ...

**Comment by romainruaud** [comment]
I agree, everything would continue to work, but since migrating to JSON is recommended, I think we should do this. It's just a small setup to write, I'll consider a PR for this one later.
